NixOS/nixpkgs
# treewide: use copyDesktopItems for desktop entry installation

## Description of changes

As the title says. Now we use `desktopItems` + `copyDesktopItems` instead of `desktopItem` + manual installation. Also fixed some dumb mistakes.

This is tested with:

```sh
grep -rl desktopItem pkgs | xargs grep -L copyDesktopItems
```

Also some foolish usage with:

```sh
grep -rl desktopItem pkgs | xargs grep -L desktopItems | xargs grep -l copyDesktopItems
```

I found that there are even some packages that do not have the defined desktop entry installed at all...

~~I admit that I am a little dizzy now, and there are still many files that need to be modified, so I will mark as draft for the time being.~~

Test guide:

1. build all packages (Sadly I don't think my laptop can finish this in a reasonable time...) and check failures
2. Diff `$out/share/applications` with the same package in last commit

## Things done



- Built on platform(s)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)



---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc
